src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs

```diff
@@ -1472,6 +1559,41 @@ public void DelaySignWithAssemblySignatureKey()
             //(rather than the contents of the keyfile or container) are used to compute the size needed to 
             //reserve in the binary for its signature. Signing using this key is only supported via sn.exe
 
+            var options = TestOptions.ReleaseDll
+                .WithDelaySign(true)
+                .WithCryptoKeyFile(s_keyPairFile)
+                .WithStrongNameProvider(s_defaultPortableProvider);
+
+            var other = CreateCompilation(
+                @"
+[assembly: System.Reflection.AssemblyDelaySign(true)]
+[assembly: System.Reflection.AssemblySignatureKey(""002400000c800000140100000602000000240000525341310008000001000100613399aff18ef1a2c2514a273a42d9042b72321f1757102df9ebada69923e2738406c21e5b801552ab8d200a65a235e001ac9adc25f2d811eb09496a4c6a59d4619589c69f5baf0c4179a47311d92555cd006acc8b5959f2bd6e10e360c34537a1d266da8085856583c85d81da7f3ec01ed9564c58d93d713cd0172c8e23a10f0239b80c96b07736f5d8b022542a4e74251a5f432824318b3539a5a087f8e53d2f135f9ca47f3bb2e10aff0af0849504fb7cea3ff192dc8de0edad64c68efde34c56d302ad55fd6e80f302d5efcdeae953658d3452561b5f36c542efdbdd9f888538d374cef106acf7d93a4445c3c73cd911f0571aaf3d54da12b11ddec375b3"", ""a5a866e1ee186f807668209f3b11236ace5e21f117803a3143abb126dd035d7d2f876b6938aaf2ee3414d5420d753621400db44a49c486ce134300a2106adb6bdb433590fef8ad5c43cba82290dc49530effd86523d9483c00f458af46890036b0e2c61d077d7fbac467a506eba29e467a87198b053c749aa2a4d2840c784e6d"")]
+public class C
+{
+  static void Goo() {}
+}",
+                new MetadataReference[] { MscorlibRef_v4_0_30316_17626 },
+                options: options);
+
+            using (var metadata = ModuleMetadata.CreateFromImage(other.EmitToArray()))
+            {
+                var header = metadata.Module.PEReaderOpt.PEHeaders.CorHeader;
+                //confirm header has expected SN signature size
+                Assert.Equal(256, header.StrongNameSignatureDirectory.Size);
+            }
+        }
```

Comment:
Consider creating a helper method shared by this test and the one below: `DelaySignWithAssemblySignatureKey(StrongNameProvider provider)`.

Reply:
Done.